src/compiler/core.ts

```diff
@@ -515,12 +525,23 @@ namespace ts {
         return result || array;
     }
 
+    export function mapAllOrFail<T, U>(array: ReadonlyArray<T>, mapFn: (x: T, i: number) => U | undefined): U[] | undefined {

```

Comment:
You can pass a function that only takes `x`, right?

Reply:
:+1: